apps/gallery/test/marionette/lib/fullscreen_view.js

```diff
@@ -0,0 +1,331 @@
+    editSaveButton: '#edit-save-button',
+    editToolApplyButton: '#edit-tool-apply-button',
+    editHeader: '#edit-view gaia-header',
+    fullscreenFrame2: '#frame2',
+    fullscreenFrame3: '#frame3',
+    cropDoneButton: '#crop-done-button',
+    editCropCanvas: '#edit-crop-canvas'
+  });
+
+  Fullscreen_View.prototype = {  
+    
+    client: null,
+
+    /**
+    * @return {Marionette.Element} Currently displayed image element.
+    */
+  	get displayedImage() {
+      return this.client.helper.waitForElement(
+        Fullscreen_View.Selector.currentImage);
+    },
+
+    /**
+    * @return {numeric} Currently displayed image width.
+    */
+  	get width() {
+  		return this.displayedImage.size().width;
+  	},
+
+    /**
+    * @return {numeric} Currently displayed image height.
+    */
+    get height() {
+    	return this.displayedImage.size().height;
+  	},
+
+    /**
+    * @return {Marionette.Element} Container for fullscreen view of an image.
+    */
+    get fullscreenView() {
+      return this.client.helper.waitForElement(
+        Fullscreen_View.Selector.fullscreenView);
+    },
+
+    /**
+    * @return {boolean} display status of Fullscreen View.
+    */
+    get displayed() {
+      return this.client.helper.waitForElement(
+        Fullscreen_View.Selector.fullscreenView).displayed();
+    },
+
+    /**
+     * @return {Marionette.Element} Element to click to get back
+     *                              to thumbnail view.
+     */
+    get fullscreenBackButton() {
+      return this.client.helper.waitForElement(
+        Fullscreen_View.Selector.fullscreenBackButton);
+    },
+
+    /**
+     * @return {Marionette.Element} Element to click for image editing mode.
+     */
+    get editButton() {
+      return this.client.helper.waitForElement(
+        Fullscreen_View.Selector.editButton);
+    },
+
+    /**
+     * @return {Marionette.Element} Element to click for effects editing mode.
+     */
+    get editEffectButton() {
+      return this.client.helper.waitForElement(
+        Fullscreen_View.Selector.editEffectButton);
+    },
+
+    /**
+     * @return {Marionette.Element} Element to click for the exposure
+     *                              editing mode.
+     */
+    get editEnhanceButton() {
+      return this.client.helper.waitForElement(
+        Fullscreen_View.Selector.editEnhanceButton);
+    },
+
+    /**
+     * @return {Marionette.Element} Element to click for the exposure editing
+     *                              mode.
+     */
+    get editExposureButton() {
+      return this.client.helper.waitForElement(
+        Fullscreen_View.Selector.editExposureButton);
+    },
+
+    /**
+     * @return {Marionette.Element} Element to click for crop editing mode.
+     */
+    get editCropButton() {
+      return this.client.helper.waitForElement(
+        Fullscreen_View.Selector.editCropButton);
+    },
+
+    /**
+     * @return {Marionette.Element} Container to host the exposure options' tab.
+     */
```

Comment:
nit: looks like typo, options' should be options. We can change comment to "Container to host the exposure options."